### PR TITLE
Add distributed series support

### LINKED ISSUE #504: Implement distributed series
**Is your feature request related to a problem? Please describe.**

We already have [distributed DataFrame](https://github.com/mars-project/mars/pull/399), we need distributed series as well.

### PR DESCRIPTION
**Pull request by qinxuye**
## What do these changes do?

This PR adds distributed Series support, tests are added for local cluster and distributed schedulers.

## Related issue number

Resolves #504 .

### CODE REVIEW

File: mars/dataframe/expressions/core.py [hunk 1]
```diff
@@ -12,34 +12,70 @@
...
+from ..core import DATAFRAME_CHUNK_TYPE, SERIES_CHUNK_TYPE, INDEX_CHUNK_TYPE, \
+    DATAFRAME_TYPE, SERIES_TYPE, INDEX_TYPE
+
+
+class ObjectType(Enum):
+    dataframe = 1
+    series = 2
+    index = 3
+
+    def __mars_tokenize__(self):
```
Comment (wjsi) [inline mars/dataframe/expressions/core.py]: Currently tokenize() can handle Enum type. This is solved in #521 .

File: mars/dataframe/expressions/core.py [hunk 2]
```diff
@@ -54,6 +90,19 @@ def new_dataframe(self, inputs, shape=None, dtypes=None, index_value=None, colum
         return self.new_dataframes(inputs, shape=shape, dtypes=dtypes,
                                    index_value=index_value, columns_value=columns_value, **kw)[0]
 
+    def new_seriess(self, inputs, shape=None, dtype=None, index_value=None, name=None,
```
Comment (wjsi) [inline mars/dataframe/expressions/core.py]: Looks a little strange for me. Maybe we can have a better name given that ``series`` itself is a plural form.

Comment (qinxuye) [reply mars/dataframe/expressions/core.py]: `series` is already plural, use `seriess` to keep consistency with others.

**Comment by qinxuye** [comment]
> LGTM except for two questions. What's more, shall we add a unified `from_pandas` to handle both pandas DFs and Series?

I have thought about that, but the `tile` are a bit different, so I separate them. Anyway, we have a `md.DataFrame` or `md.Series` as the user entry point to create a Mars df or series from pandas one.

**Pull request merged**